utilities/titandb/titan_db_test.cc

```diff
@@ -388,23 +388,20 @@ TEST_F(TitanDBTest, BlobFileIOError) {
   for(auto& it : data) {
     std::string value;
     if(it.second.size() > options_.min_blob_size) {
-      ASSERT_FALSE(db_->Get(ReadOptions(), it.first, &value).ok());
+      ASSERT_TRUE(db_->Get(ReadOptions(), it.first, &value).IsIOError());
       mock_env->SetFilesystemActive(true);
     }
   }
   std::unique_ptr<Iterator> iter(db_->NewIterator(ReadOptions()));
   iter->SeekToFirst();
   mock_env->SetFilesystemActive(true);
-  for (const auto& it : data) {
-    if(iter->Valid()) {
-      ASSERT_EQ(it.first, iter->key());
-      ASSERT_EQ(it.second, iter->value());
-      ASSERT_TRUE(it.second.size() < options_.min_blob_size);
-    } else {
-      ASSERT_TRUE(it.second.size() >= options_.min_blob_size);
-      break;
-    }
+  if(iter->Valid()) {
```

Comment:
just assert its invalid here.

And also test the case: setting `SetFilesystemActive=true` before `SeekToFirst` (you need to disable the sync point callback), then before `Next` set `SetFilesystemActive=false`.

Same for `BlobFileCorruptionErrorHandling` test below. You need to test both when `Seek` get error and when `Next` get error.

Reply:
But `SeekToFirst` shouldn't be invalid since the deactivation is only triggered when trying to read BlobFile.
Or do you mean we simply test overall filesystem error here?

Reply:
Only put long values will satisfy the need.

Reply:
`SeekToFirst` should read the first key from blob file.

Reply:
oh, sorry, the first key is always a inlined key. In that case, just assert its valid here. And let's also test the case `Seek(second_key)` and it should invalidate immediately.

Reply:
my bad, `Put` data starts from k=1, so the SeekFirst should indeed be invalid.